docs/VLSI/Sky130-Tutorial.rst

```diff
@@ -0,0 +1,157 @@
+
+
+example-sky130.yml
+^^^^^^^^^^^^^^^^^^
+This contains the Hammer configuration for this example project. Example clock constraints, power straps definitions, placement constraints, and pin constraints are given. Additional configuration for the extra libraries and tools are at the bottom.
+
+First, set ``technology.sky130.sky130A`` to the absolute path to the ``sky130A`` directory containing the Sky130 PDK files. See the 
+`Sky130 Hammer plugin README  <https://github.com/ucb-bar/hammer/blob/sky130sram/src/hammer-vlsi/technology/sky130/README.md>`__
+for details about the PDK setup.
+
+
+Synthesis
+^^^^^^^^^
+.. code-block:: shell
+
+    make syn tech_name=sky130 CONFIG=TinyRocketConfig
+
+Post-synthesis logs and collateral are in ``build/syn-rundir``. The raw QoR data is available at ``build/syn-rundir/reports``, and methods to extract this information for design space exploration are a WIP.
+
+Place-and-Route
+^^^^^^^^^^^^^^^
+.. code-block:: shell
+
+    make par tech_name=sky130 CONFIG=TinyRocketConfig
+
+After completion, the final database can be opened in an interactive Innovus session via ``./build/par-rundir/generated-scripts/open_chip``.
+
+Intermediate database are written in ``build/par-rundir`` between each step of the ``par`` action, and can be restored in an interactive Innovus session as desired for debugging purposes. 
+
+Timing reports are found in ``build/par-rundir/timingReports``. They are gzipped text files.
+
+DRC & LVS
+^^^^^^^^^
+To run DRC & LVS, and view the results in Calibre:
+
+.. code-block:: shell
+
+    make drc tech_name=sky130 CONFIG=TinyRocketConfig
+    ./build/chipyard.TestHarness.TinyRocketConfig-ChipTop/drc-rundir/generated-scripts/view_drc
+    make lvs tech_name=sky130 CONFIG=TinyRocketConfig
+    ./build/chipyard.TestHarness.TinyRocketConfig-ChipTop/lvs-rundir/generated-scripts/view_lvs
+
+Some DRC errors are expected from this PDK, especially with regards to the SRAMs, as explained in the 
+`Sky130 Hammer plugin README  <https://github.com/ucb-bar/hammer/blob/master/src/hammer-vlsi/technology/sky130/README.md>`__.
+For this reason, the ``example-vlsi-sky130`` script black-boxes the SRAMs for DRC/LVS analysis. 
+
+Simulation
+^^^^^^^^^^
+Simulation with VCS is supported, and can be run at the RTL- or gate-level (post-synthesis and post-P&R). The simulation infrastructure as included here is intended for running RISC-V binaries on a Chipyard config. For example, for an RTL-level simulation:
+
+.. code-block:: shell
+
+    make sim-rtl CONFIG=TinyRocketConfig BINARY=$RISCV/riscv64-unknown-elf/share/riscv-tests/isa/rv64ui-p-simple
+
+Post-synthesis and post-P&R simulations use the ``sim-syn`` and ``sim-par`` make targets, respectively.
+
+Appending ``-debug`` and ``-debug-timing`` to these make targets will instruct VCS to write a SAIF + VPD and do timing-annotated simulations, respectively. See the ``sim.mk`` file for all available targets.
```

Comment:
Isn't VPDs deprecated due to DVE being changed to use FSDBs? https://github.com/ucb-bar/chipyard/pull/1072

Reply:
Technically, no, it's that DVE is no longer available due to licensing, but VCS can still output VPDs.

Perhaps this can be updated to say something like "or FSDB if the USE_FSDB flag is set"

Reply:
I like this better (but would prefer to remove the vpd reference even more).

Reply:
Just added the FSDB note, I'll leave VPD for now but fine with removing it if that's better.